magda-authorization-api/src/test/mockDatabase.ts

```diff
@@ -0,0 +1,53 @@
+import mockUserDataStore from "@magda/typescript-common/dist/test/mockUserDataStore";
+import { User } from "@magda/typescript-common/dist/authorization-api/model";
+import { Maybe } from "tsmonad";
+import arrayToMaybe from "@magda/typescript-common/dist/util/arrayToMaybe";
+
+export default class MockDatabase {
+    getUser(id: string): Promise<Maybe<User>> {
+        return new Promise(function(resolve, reject) {
+            resolve(
+                arrayToMaybe(
+                    mockUserDataStore.getRecordByUserId(id).map(item => ({
+                        id: item.id,
```

Comment:
Are you trying to clone the item here? You could use `item => ({...item})` I think?

Reply:
Notice your Database.ts return different set of fields from two different method (one method will skip field `sourceId`) --- thought you want to make sure different methods have different access and trying to simulate the same behaviour --- although I think I forgot to remove the `sourceId` from the second mocked method.